File: 2998-count-symmetric-integers/2998-count-symmetric-integers.ts
function countSymmetricIntegers(low: number, high: number): number {
    let cnt = 0;

    for (let num = low; num <= high; num++) {
        let str = num.toString();
        const len = str.length;

        if (len % 2 != 0) continue;

        let half = len / 2;
        let leftSum = 0;
        let rightSum = 0;

        for (let i = 0; i < half; i++) {
            leftSum += parseInt(str[i]);
        }
        for (let i = half; i < len; i++) {
            rightSum += parseInt(str[i]);
        }
        if (leftSum === rightSum) {
            cnt++;
        }
    }
    return cnt;
};

File: 2998-count-symmetric-integers/2998-count-symmetric-integers.cpp
class Solution {
public:
    int countSymmetricIntegers(int low, int high) {
        int cnt = 0;
        for (int num = low; num <= high; num++) {
            string str = to_string(num);
            short len = str.length();

            if (len % 2 == 1)
                continue;

            short half = len / 2;
            int leftSum = 0;
            int rightSum = 0;

            for (int i = 0; i < half; i++) {
                leftSum += str[i] - '0';
                rightSum += str[i + half] - '0';
            }

            if (leftSum == rightSum)
                cnt++;
        }
        return cnt;
    }
};